cmake/modules/RootConfiguration.cmake

```diff
@@ -755,7 +755,8 @@ if(WIN32)
   # We cannot use the compiledata.sh script for windows
   configure_file(${CMAKE_SOURCE_DIR}/cmake/scripts/compiledata.win32.in ${CMAKE_BINARY_DIR}/include/compiledata.h NEWLINE_STYLE UNIX)
 else()
-  execute_process(COMMAND ${CMAKE_SOURCE_DIR}/build/unix/compiledata.sh ${CMAKE_BINARY_DIR}/include/compiledata.h "${CXX}"
+  execute_process(COMMAND ${CMAKE_SOURCE_DIR}/build/unix/compiledata.sh
+    ${CMAKE_BINARY_DIR}/include/compiledata.h "${CMAKE_CXX_COMPILER}"
```

Comment:
Unless I am mistaken, this will break the ability of running ACLiC if the compiler in the run-time environment is not the exact same place as the compiler was when ROOT was built.

Reply:
As I say in the commit message, within `compiledata.sh`, the basename is used in any case. Everything behaves exactly the same as before. I tried with a compiler not in my `$PATH` to check.

Reply:
Fair enough.  I had forgotten about that part of compiledata.h (and did not notice the commit message).  Thanks.

Reply:
I think I will soon update how `compiledata.h` is generated in any case. The shell script is ugly. Now that the configure build system is no longer there, we can probably get away with a simple `configure_file()` call from CMake, like we do for Windows.